components/nimbus/src/stateful/targeting.rs

```diff
@@ -27,3 +36,82 @@ impl NimbusTargetingHelper {
     }
 }
+
+pub trait RecordedContext: Send + Sync {
+    /// Returns a JSON representation of the context object
+    ///
+    /// This method will be implemented in foreign code, i.e: Kotlin, Swift, Python, etc...
+    fn to_json(&self) -> JsonObject;
+
+    /// Returns a HashMap representation of the event queries that will be used in the targeting
+    /// context
+    ///
+    /// This method will be implemented in foreign code, i.e: Kotlin, Swift, Python, etc...
+    fn get_event_queries(&self) -> JsonObject;
+
+    /// Sets the object's internal value for the event query values
+    ///
+    /// This method will be implemented in foreign code, i.e: Kotlin, Swift, Python, etc...
+    fn set_event_query_values(&self, json: JsonObject);
+
+    /// Records the context object to Glean
+    ///
+    /// This method will be implemented in foreign code, i.e: Kotlin, Swift, Python, etc...
+    fn record(&self);
+}
+
+impl dyn RecordedContext {
+    pub fn execute_queries(
+        &self,
+        nimbus_targeting_helper: &NimbusTargetingHelper,
+    ) -> Result<Map<String, Value>> {
+        let results = Map::from_iter(self.get_event_queries().iter().filter_map(
+            |(key, value): (&String, &Value)| match value.as_str() {
+                Some(v) => match nimbus_targeting_helper.evaluate_jexl_raw_value(v.into()) {
+                    Ok(result) => Some((key.to_string(), result)),
+                    Err(err) => {
+                        log::info!(
+                            "error during jexl evaluation for query {} — {}",
+                            value,
+                            err.to_string()
+                        );
+                        None
+                    }
+                },
+                None => {
+                    log::info!("value {} for key {} is not a string", key, value);
+                    Some((key.to_string(), json!(value)))
+                }
+            },
+        ));
+        self.set_event_query_values(results.clone());
+        Ok(results)
+    }
+
+    pub fn validate_queries(&self) -> Result<()> {
+        for (_, value) in self.get_event_queries().iter() {
+            match value.as_str() {
+                Some(v) => match EventQueryType::validate_query(v) {
+                    Ok(result) => match result {
+                        true => {
+                            continue;
+                        }
+                        false => {
+                            return Err(NimbusError::BehaviorError(
+                                BehaviorError::EventQueryParseError(v.to_string()),
+                            ))
+                        }
+                    },
+                    Err(e) => return Err(e),
+                },
+                None => {
+                    return Err(NimbusError::BehaviorError(BehaviorError::TypeError(
+                        value.to_string(),
+                        "string".into(),
+                    )))
+                }
```

Comment:
`match` on bool is less idiomatic than if/else

```rust
Ok(result) => if result {
    continue;
} else {
    return Err(NimbusError::BehaviorError(BehaviorError::EventQueryParseError(v.to_string())));
}                       }
```

but its even more idiomatic to match on the value

```rust
Ok(true) => continue,
Ok(false) => return Err(NimbusError::BehaviorError(BehaviorError::EventQueryParseError(v.to_string()))),
```

but we can also clean this up into a single match
```rust
            match value.as_str().map(EventQueryType::validate_query) {
                Some(Ok(true)) => continue,
                Some(Ok(false)) => return Err(NimbusError::BehaviorError(
                                       BehaviorError::EventQueryParseError(v.to_string()),
                                   )),
                Some(Err(e)) => return Err(e),
                None => {
                    return Err(NimbusError::BehaviorError(BehaviorError::TypeError(
                        value.to_string(),
                        "string".into(),
                    )))
                }
```

and finally we can use some Option and result methods (plus ?) to turn this into a linear chain of calls:

```suggestion
            let query = value
                .as_str()
                .ok_or_else(|| NimbusError::BehaviorError(BehaviorError::TypeError(
                    value.into(),
                    "string".into(),
                ))?;
            
            if !EventQueryType::validate_query(query)? {
                return Err(NimbusError::BehaviorError(
                    BehaviorError::EventQueryParseError(v.into()),
                ));
            }
```